java-checks/src/main/java/org/sonar/java/checks/security/CookieShouldNotContainSensitiveDataCheck.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.security;
+
+import java.util.Arrays;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.java.checks.methods.AbstractMethodDetection;
+import org.sonar.java.matcher.MethodMatcher;
+import org.sonar.java.matcher.TypeCriteria;
+
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.NewClassTree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+
+@Rule(key = "S2255")
+public class CookieShouldNotContainSensitiveDataCheck extends AbstractMethodDetection {
+
+  private static final String SERVLET_COOKIE = "javax.servlet.http.Cookie";
+  private static final String NET_HTTP_COOKIE = "java.net.HttpCookie";
+  private static final String JAX_RS_COOKIE = "javax.ws.rs.core.Cookie";
+
+  private static final int VALUE_PARAM_INDEX = 1;
+
+  private static final String MESSAGE = "If the data stored in this cookie is sensitive, it should be stored internally in the user session.";
+
+  @Override
+  protected List<MethodMatcher> getMethodInvocationMatchers() {
+    return Arrays.asList(
+        MethodMatcher.create().typeDefinition(TypeCriteria.subtypeOf(SERVLET_COOKIE)).name("<init>").withAnyParameters(),
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define a constant instead of duplicating this literal "<init>" 3 times. ([squid:S1192](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&pullRequest=2018&issues=AWNep72O0C6aA6zgZXLY&open=AWNep72O0C6aA6zgZXLY)

[ISSUE_KEY]: AWNep72O0C6aA6zgZXLY